fwbot/models/merge_request.py

```diff
@@ -0,0 +1,313 @@
+import logging
+import re
+from typing import Any, Dict, List, Sequence, Tuple
+
+from odoo import api, fields, models
+from requests import Response
+
+from ..api import ForwardbotGitlabClient
+from .repository import Repository, multi
+
+_logger = logging.getLogger(__name__)
+_footer_regex = re.compile(r"This is an automatic forward port for !\d+$")
+
+
+class MergeRequest(models.Model):
+    _name = "fwbot.merge.request"
+    _description = "Merge Request (fwbot)"
+    _inherit = "fwbot.remote.mixin"
+
+    name = fields.Char(help="Merge Request Title", required=True)
+    description = fields.Text()
+    global_id = fields.Integer(help="ID for this MR on the remote server. Unique across all projects")
+    internal_id = fields.Integer(help="Internal ID. Unique inside the target project.")
+    source_branch = fields.Char(required=True)
+    source_repository_id: Repository = fields.Many2one("fwbot.repository", required=True)
+    target_branch = fields.Char(required=True)
+    target_repository_id: Repository = fields.Many2one("fwbot.repository", required=True)
+    state = fields.Selection(
+        [
+            ("pending", "Pending"),
+            ("opened", "Opened"),
+            ("closed", "Closed"),
+            ("locked", "Locked"),
+            ("merged", "Merged"),
+        ],
+        default="pending",
+        required=True,
+    )
+    forward_port_id: "MergeRequest" = fields.Many2one("fwbot.merge.request")
+    processed = fields.Boolean(default=False)
+
+    _sql_constraints = [
+        (
+            "unique_global_id",
+            "UNIQUE (global_id)",
+            "Two MRs can't have the same ID",
+        ),
+    ]
+
+    @multi
+    def create_forward_port(self) -> "MergeRequest":
+        self.ensure_one()
+        if self.forward_port_id:
+            return self.forward_port_id
+
+        forward_branch = self.target_repository_id.get_next_branch(self.target_branch)
+        if not forward_branch:
+            raise ValueError("can't create forward port, missing forward branch")
```

Comment:
This is a lint

```suggestion
            raise ValueError(_("can't create forward port, missing forward branch"))
```

Same in similar cases.

Reply:
I think it only applies to Odoo exceptions, I will change it to ValidationError and translate it.